src/pal/src/include/pal/sharedmemory.h

```diff
@@ -0,0 +1,247 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef _PAL_SHARED_MEMORY_H_
+#define _PAL_SHARED_MEMORY_H_
+
+#include "corunix.hpp"
+
+#ifndef static_assert_no_msg
+#define static_assert_no_msg( cond ) static_assert( cond, #cond )
+#endif // !static_assert_no_msg
+
+#ifndef _countof
+#define _countof(a) (sizeof(a) / sizeof(a[0]))
+#endif // !_countof
+
+#ifndef PAGE_SIZE
+#define PAGE_SIZE (0x1000)
+#endif // !PAGE_SIZE
```

Comment:
It's hard-coded everywhere else. I had to redefine it only here because the header it's defined in is not includable in PAL (it's in common headers that are only allowed for use in VM as far as I understand). I think there are currently several assumptions on the size of a page, and it's probably ok.